python/ray/data/dataset.py

```diff
@@ -2746,156 +2746,51 @@ def make_generator():
-                a two-element tuple containing a ``FeatureTypeSpec`` and
-                ``tf.TypeSpec`` object corresponding to (features, label). Otherwise, a
-                single ``TensorflowFeatureTypeSpec`` corresponding to features tensor.
-                A ``TensorflowFeatureTypeSpec`` is a ``tf.TypeSpec``,
-                ``List["tf.TypeSpec"]``, or ``Dict[str, "tf.TypeSpec"]``.
-            label_column: The name of the column used as the label
-                (second element of the output tuple). If not specified, output
-                will be just one tensor instead of a tuple.
-            feature_columns: The names of the columns to use as the features. Can be a
-                list of lists or a dict of string-list pairs for multi-tensor output.
-                If None, then use all columns except the label columns as the features.
-            prefetch_blocks: The number of blocks to prefetch ahead of the
-                current block during the scan.
-            batch_size: Record batch size. Defaults to 1.
-            drop_last: Set to True to drop the last incomplete batch,
-                if the dataset size is not divisible by the batch size. If
-                False and the size of dataset is not divisible by the batch
-                size, then the last batch will be smaller. Defaults to False.
-            local_shuffle_buffer_size: If non-None, the data will be randomly shuffled
-                using a local in-memory shuffle buffer, and this value will serve as the
-                minimum number of rows that must be in the local in-memory shuffle
-                buffer in order to yield a batch. When there are no more rows to add to
-                the buffer, the remaining rows in the buffer will be drained. This
-                buffer size must be greater than or equal to ``batch_size``, and
-                therefore ``batch_size`` must also be specified when using local
-                shuffling.
-            local_shuffle_seed: The seed to use for the local random shuffle.
-
-        Returns:
-            A tf.data.Dataset.
-        """
-
-        # argument exception checking is done in from_generator
-
-        try:
-            import tensorflow as tf
-        except ImportError:
-            raise ValueError("tensorflow must be installed!")
-
-        from ray.air._internal.tensorflow_utils import convert_pandas_to_tf_tensor
-
-        # `output_signature` can be a tuple but not a list. See
-        # https://stackoverflow.com/questions/59092423/what-is-a-nested-structure-in-tensorflow.
-        if isinstance(output_signature, list):
-            output_signature = tuple(output_signature)
-
-        def make_generator():
-            for batch in self.iter_batches(
+        from ray.air._internal.tensorflow_utils import get_type_spec
+        from ray.train.tensorflow import prepare_dataset_shard
+
+        if self._dataset_format() == "simple":
+            raise NotImplementedError
```

Comment:
what about "simple" makes it not able to be supported?

Reply:
is this because of the schema?

Reply:
`feature_columns` and `label_columns` doesn't make sense for simple datasets; simple datasets don't have columns. Also, it's hard to infer data types from unstructured data.

We could try to infer shape and data types from a list of objects, but I suspect that'd be messy.